### PR TITLE
Microsoft teams

### LINKED ISSUE #585: [FEATURE] Support Microsoft Teams webhooks
**Is your feature request related to a problem?**
Posting to MSTeams channel through incoming webhook connector

**What solution would you like?**
New channel

**What alternatives have you considered?**
-

**Do you have any additional context?**
The message has to be wrapped in a json object, in it's most basic form it only contains a "text" property:

curl -H 'Content-Type: application/json' -d '{"text": "Hello World"}' <YOUR WEBHOOK URL>

### PR DESCRIPTION
**Pull request by zhichao-aws**
### Description
add Microsoft Teams as notification channel type
### Issues Resolved
#585 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by zhichao-aws** [comment]
PR in common-utils: https://github.com/opensearch-project/common-utils/pull/428

**Comment by zhichao-aws** [comment]
tests depending on the PR in common-utils https://github.com/opensearch-project/common-utils/pull/428

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt [hunk 1]
```diff
@@ -159,12 +160,14 @@ class DestinationHttpClient {
         val keyName = when (destination) {
             // Slack webhook request body has required "text" as key name https://api.slack.com/messaging/webhooks
             // Chime webhook request body has required "Content" as key name
+            // Microsoft Teams webhook request body has required "text" as key name https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors
```
Comment (xluo-aws) [inline notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt:163]: Should the url be https://learn.microsoft.com/en-us/microsoftteams/platform/webhooks-and-connectors/what-are-webhooks-and-connectors?

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/ConfigIndexingActions.kt [hunk 2]
```diff
@@ -65,6 +64,11 @@ object ConfigIndexingActions {
         // TODO: URL validation with rules
     }
 
+    @Suppress("UnusedPrivateMember")
+    private fun validateMicrosoftTeamsConfig(microsoftTeams: MicrosoftTeams, user: User?) {
+        // TODO: host validation with rules
```
Comment (xluo-aws) [inline notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/ConfigIndexingActions.kt:69]: Nothing to validate?

File: notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt (hunk 1 above)
Comment (zhichao-aws) [reply notifications/core/src/main/kotlin/org/opensearch/notifications/core/client/DestinationHttpClient.kt:163]: Yes, you're right

File: notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/ConfigIndexingActions.kt (hunk 2 above)
Comment (zhichao-aws) [reply notifications/notifications/src/main/kotlin/org/opensearch/notifications/index/ConfigIndexingActions.kt]: Add some logic to make sure https and "webhook.office.com" in the url. But we still lack validation for existing channels such as chime and slack. Maybe we can do some research and add more validation for them in another PR.

**Review by xluo-aws — DISMISSED** [review]

**Review by xluo-aws — APPROVED** [review]

**Review by yuye-aws — COMMENTED** [review]
Looks good to me!

**Review by yuye-aws — APPROVED** [review]

**Review by gaobinlong — APPROVED** [review]
LGTM.

**Pull request merged**